### ISSUE TITLE
Segment task

### ISSUE THREAD

**Issue by NSonTung463** [THU-MIG/yolov10#30]
When will team release task segmentation?

**Comment by alexkutsan** [comment]
Waiting for this feature as well.

**Comment by TheYoungBeast** [comment]
It seems YOLOv10 supports segmentation

> Thanks for your interest! Achieving the task of rotating target detection and segmentation is valuable. While there are no corresponding pre-trained weights yet, YOLOv10 can support these tasks like YOLOv8.

_Originally posted by @jameslahm in https://github.com/THU-MIG/yolov10/issues/4#issuecomment-2128566348_

**Comment by lijoe123** [comment]
> It seems YOLOv10 supports segmentation
> 
> > Thanks for your interest! Achieving the task of rotating target detection and segmentation is valuable. While there are no corresponding pre-trained weights yet, YOLOv10 can support these tasks like YOLOv8.
> 
> _Originally posted by @jameslahm in [#4 (comment)](https://github.com/THU-MIG/yolov10/issues/4#issuecomment-2128566348)_

Hey bro! Do you know how to use the yolov10 segmentation?

**Comment by TheYoungBeast** [comment]
> > It seems YOLOv10 supports segmentation
> > > Thanks for your interest! Achieving the task of rotating target detection and segmentation is valuable. While there are no corresponding pre-trained weights yet, YOLOv10 can support these tasks like YOLOv8.
> > 
> > 
> > _Originally posted by @jameslahm in [#4 (comment)](https://github.com/THU-MIG/yolov10/issues/4#issuecomment-2128566348)_
> 
> Hey bro @TheYoungBeast ! Do you know how to use the yolov10 segmentation?

It is not implemented in ultralytics yet, I think.
For example - there is only a loss function for ``detect`` task in YOLOv10 (https://github.com/THU-MIG/yolov10/blob/main/ultralytics/utils/loss.py#L717)
You would have to implement ``v8SegmentationLoss`` function equivalent for YOLOv10 and override ``task_map`` method to return 'segment' task.
You can try to reuse the v8 methods:
```py
class YOLOv10(Model, ....)
 @property
    def task_map(self):
        """Map head to model, trainer, validator, and predictor classes."""
        return {
            "detect": {
                "model": YOLOv10DetectionModel,
                "trainer": YOLOv10DetectionTrainer,
                "validator": YOLOv10DetectionValidator,
                "predictor": YOLOv10DetectionPredictor,
            },
            "segment": {
                "model": SegmentationModel,
                "trainer": yolo.segment.SegmentationTrainer,
                "validator": yolo.segment.SegmentationValidator,
                "predictor": yolo.segment.SegmentationPredictor,
            },  
        }
```

It will start the training process for segmentation (task), although it will crash during the validation process (because v8 loss function seems to be not compatible with v10)

I might be wrong on this topic, I have little to no experience with this framework/library...
